virtwho/executor.py

```diff
@@ -68,8 +74,36 @@ def _create_virt_backends(self):
                 self.logger.error('Unable to use configuration "%s": %s', name, str(e))
                 continue
             virts.append(virt)
+        self._init_run_status(config_names)
         return virts
 
+    def _init_run_status(self, config_names=[]):
+        # ensure that there is an entry in the status file for each config. We don't care if they work, we
+        # need to record their existence
+        lock = PIDLock(STATUS_LOCK)
+        while lock.is_locked():
+            time.sleep(1)
+        with lock:
+            os.makedirs(STATUS_DATA_DIR, exist_ok=True)
+            # need to create the file if it does not exist
+            with open(STATUS_DATA, "w+") as json_status:
+                try:
+                    status_dict = json.load(json_status)
+                except JSONDecodeError:
+                    status_dict = {}
+            if 'sources' not in status_dict:
+                status_dict['sources'] = {}
+            if 'destinations' not in status_dict:
+                status_dict['destinations'] = {}
+            for name in config_names:
+                if name not in status_dict['sources']:
+                    status_dict['sources'][name] = {"last_successful_retrieve": "none"}
```

Comment:
I believe that the value should be `None` and not "none", because we want to store `null` value to JSON file, right? Or do we want to store "none" string there?